File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/1730_판화/1730.java
import java.util.Scanner;

class Main
{
    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        String command = sc.hasNext() ? sc.next() : "";

        boolean[][] passVertical = new boolean[N][N];
        boolean[][] passHorizontal = new boolean[N][N];

        int curR = 0, curC = 0;
        for (int i = 0; i < command.length(); i++) {
            char cmd = command.charAt(i);
            if (cmd == 'D') {
                if (curR == N - 1) continue;;
                passVertical[curR][curC] = passVertical[curR + 1][curC] = true;
                curR++;
            }
            else if (cmd == 'U') {
                if (curR == 0) continue;
                passVertical[curR][curC] = passVertical[curR - 1][curC] = true;
                curR--;
            }
            else if (cmd == 'L') {
                if (curC == 0) continue;;
                passHorizontal[curR][curC] = passHorizontal[curR][curC - 1] = true;
                curC--;
            }
            else {
                if (curC == N - 1) continue;;
                passHorizontal[curR][curC] = passHorizontal[curR][curC + 1] = true;
                curC++;
            }
        }

        for (int i = 0; i < N; i++) {
            String ans = "";
            for (int j = 0; j < N; j++) {
                if (passHorizontal[i][j] && passVertical[i][j]) ans += "+";
                else if (passVertical[i][j]) ans += "|";
                else if (passHorizontal[i][j]) ans += "-";
                else ans += ".";
            }
            System.out.println(ans);
        }
    }
}

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/1730_판화/1730.py
N = int(input())
command = input()
pass_vertical = [[0 for _ in range(N)] for _ in range(N)]
pass_horizontal = [[0 for _ in range(N)] for _ in range(N)]

x, y = 0, 0
for i in command:
    if i == 'U':
        if y > 0:
            pass_horizontal[y][x] = True
            y -= 1
            pass_horizontal[y][x] = True
    elif i == 'D':
        if y < N - 1:
            pass_horizontal[y][x] = True
            y += 1
            pass_horizontal[y][x] = True
    elif i == 'L':
        if x > 0:
            pass_vertical[y][x] = True
            x -= 1
            pass_vertical[y][x] = True
    elif i == 'R':
        if x < N - 1:
            pass_vertical[y][x] = True
            x += 1
            pass_vertical[y][x] = True

for i in range(N):
    for j in range(N):
        if not pass_vertical[i][j] and not pass_horizontal[i][j]:
            print(".", end="")
        elif not pass_vertical[i][j] and pass_horizontal[i][j]:
            print("|", end="")
        elif pass_vertical[i][j] and not pass_horizontal[i][j]:
            print("-", end="")
        else:
            print("+", end="")
    print("")
